pkg/controllermanager/controller/shoot/state/finalizer/reconciler.go

```diff
@@ -0,0 +1,90 @@
+// SPDX-FileCopyrightText: SAP SE or an SAP affiliate company and Gardener contributors
+//
+// SPDX-License-Identifier: Apache-2.0
+
+package finalizer
+
+import (
+	"context"
+	"fmt"
+
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+	logf "sigs.k8s.io/controller-runtime/pkg/log"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+
+	gardencorev1beta1 "github.com/gardener/gardener/pkg/apis/core/v1beta1"
+	controllermanagerconfigv1alpha1 "github.com/gardener/gardener/pkg/controllermanager/apis/config/v1alpha1"
+	"github.com/gardener/gardener/pkg/controllerutils"
+)
+
+// FinalizerName is the finalizer used by the controller.
+const FinalizerName = "core.gardener.cloud/shootstate"
+
+// Reconciler reconciles ShootState objects and ensures the finalizer
+// exists during "Migrate" and "Restore" phases of Shoot migration to
+// another Seed.
+type Reconciler struct {
+	// Client is the API Server client used by the Reconciler.
+	Client client.Client
+	Config controllermanagerconfigv1alpha1.ShootStateControllerConfiguration
```

Comment:
Add doc string for the `Config` field.

Reply:
Thanks for catching this missing doc string, added it in the latest updates.